tests/test_requests.py

```diff
@@ -81,6 +81,62 @@ async def test_http_client_single_requests(async_mock):
         ]
 
 
+@pytest.fixture
+def client_with_status():
+    def _param_wrapper(status_code: int):
+        async def stub_request_downloader(*args, **kwargs):
+            async def stub(req):
+                return HttpResponse(req.url, body=b"", status=status_code)
+            return await stub(*args, **kwargs)
+        return stub_request_downloader
+    return _param_wrapper
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("method_name", ["request", "get", "post", "execute"])
+async def test_http_client_allow_status(async_mock, client_with_status, method_name):
+    client = HttpClient(async_mock)
+
+    # Simulate 500 Internal Server Error responses
+    client._request_downloader = client_with_status(500)
+
+    method = getattr(client, method_name)
+
+    url_or_request = "url"
+    if method_name == "execute":
+        url_or_request = HttpRequest(url_or_request)
+
+    # Should handle single and multiple values
+    await method(url_or_request, allow_status=500)
+    response = await method(url_or_request, allow_status=[500, 503])
+    assert isinstance(response, HttpResponse)
+    assert response.status == 500
+
+    # As well as strings
+    await method(url_or_request, allow_status="500")
+    await method(url_or_request, allow_status=["500", "503"])
+
+    with pytest.raises(HttpResponseError) as excinfo:
+        await method(url_or_request)
+    assert isinstance(excinfo.value.request, HttpRequest)
+    assert isinstance(excinfo.value.response, HttpResponse)
```

Comment:
@kmike could you refresh my memory on why the `HttpRequest` and `HttpResponse` classes were set using `@attr.define(..., eq=False)`? [link to code](https://github.com/scrapinghub/web-poet/blob/master/web_poet/page_inputs.py#L148)

Setting `eq=True` could allow easy comparisons here in the test.